File: ios/music/music/Playlist.swift
import Foundation

class Playlist: Codable, Identifiable {
    var id: String
    var name: String
    var image: String?
    var songs: [Song]?
    
    enum CodingKeys: String, CodingKey {
        case id, name, songs
        case image = "image_url"
    }
}

File: app/Models/Playlist.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Playlist extends Model
{
    protected $fillable = [
        'name',
        'image_url',
    ];

    public function songs()
    {
        return $this->belongsToMany(Song::class, 'playlist_songs', 'playlist_id', 'song_isrc', 'id', 'isrc')->withTimestamps();
    }
}
